File: OBMSAPP-API/OBMS.WebAPI/Models/Domain/EmployeeVisaPassportListView.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace OBMS.WebAPI.Models.Domain
{
    /// <summary>
    /// Result model for Employee Visa & Passport List report.
    /// Populated via raw SQL query — no backing table.
    /// </summary>
    public class EmployeeVisaPassportListView
    {
        [Key]
        public int EMP_ID { get; set; }
        public string EMP_CODE { get; set; } = string.Empty;
        public string EMP_NAME { get; set; } = string.Empty;
        public string EMP_BRANCH_CODE { get; set; } = string.Empty;
        public string EMP_PASSPORT_NO { get; set; } = string.Empty;
        public string EMP_IC_NEW { get; set; } = string.Empty;
        public DateTime? VisaExpiryDate { get; set; }
        public DateTime? PassportExpiryDate { get; set; }
        public DateTime? EMPPAY_DATE_JOINED { get; set; }
        public string? EMPPAY_CATEGORY { get; set; }
        public int? DaysToVisaExpiry { get; set; }
        public int? DaysToPassportExpiry { get; set; }
        public string VisaStatus { get; set; } = string.Empty;
        public string PassportStatus { get; set; } = string.Empty;
    }
}

File: OBMSAPP-UI/angularapp/src/app/model/EmployeeVisaPassportListView.ts
export class EmployeeVisaPassportListView {
  EMP_ID: number;
  EMP_CODE: string;
  EMP_NAME: string;
  EMP_BRANCH_CODE: string;
  EMP_PASSPORT_NO: string;
  EMP_IC_NEW: string;
  VisaExpiryDate: string | null;
  PassportExpiryDate: string | null;
  EMPPAY_DATE_JOINED: string | null;
  EMPPAY_CATEGORY: string;
  DaysToVisaExpiry: number | null;
  DaysToPassportExpiry: number | null;
  VisaStatus: string;
  PassportStatus: string;

  constructor(data: Partial<EmployeeVisaPassportListView> = {}) {
    this.EMP_ID             = data.EMP_ID             ?? 0;
    this.EMP_CODE           = data.EMP_CODE           ?? '';
    this.EMP_NAME           = data.EMP_NAME           ?? '';
    this.EMP_BRANCH_CODE    = data.EMP_BRANCH_CODE    ?? '';
    this.EMP_PASSPORT_NO    = data.EMP_PASSPORT_NO    ?? '';
    this.EMP_IC_NEW         = data.EMP_IC_NEW         ?? '';
    this.VisaExpiryDate     = data.VisaExpiryDate     ?? null;
    this.PassportExpiryDate = data.PassportExpiryDate ?? null;
    this.EMPPAY_DATE_JOINED = data.EMPPAY_DATE_JOINED ?? null;
    this.EMPPAY_CATEGORY    = data.EMPPAY_CATEGORY    ?? '';
    this.DaysToVisaExpiry   = data.DaysToVisaExpiry   ?? null;
    this.DaysToPassportExpiry = data.DaysToPassportExpiry ?? null;
    this.VisaStatus         = data.VisaStatus         ?? '';
    this.PassportStatus     = data.PassportStatus     ?? '';
  }
}
